pkg/provider/app_oauth_resource.go

```diff
@@ -0,0 +1,241 @@
+		)
+	}
+}
+
+func (r *AppOAuthResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data AppOAuthResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	request := &client.CreateAppOAuthProviderRequest{
+		Name:         data.Name.ValueString(),
+		Type:         data.Type.ValueString(),
+		ClientID:     data.ClientID.ValueString(),
+		ClientSecret: data.ClientSecret.ValueString(),
+	}
+
+	traceAPICall("CreateAppOAuthProvider")
+	createResp, err := r.provider.service.CreateAppOAuthProvider(ctx, request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating OAuth provider",
+			fmt.Sprintf("Could not create OAuth provider: %s", err),
+		)
+		return
+	}
+	data.ID = types.StringValue(createResp.ID)
+	data.SecureConfigID = types.StringValue(createResp.SecureConfigID)
+	data.OrgID = types.StringValue(createResp.OrgID)
+	data.Status = types.StringValue(createResp.Status)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *AppOAuthResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data AppOAuthResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.ID.IsNull() {
+		return
+	}
+
+	traceAPICall("GetAppOAuthProvider")
+	getResp, err := r.provider.service.GetAppOAuthProvider(ctx, data.ID.ValueString())
+	if err != nil {
+		if _, ok := err.(*client.NotFoundError); ok {
+			resp.Diagnostics.AddWarning(
+				"OAuth provider not found",
+				fmt.Sprintf("OAuth provider with ID %s is not found. Removing from state.", data.ID.ValueString()))
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Error getting OAuth provider with ID %s", data.ID.ValueString()),
+				err.Error())
+		}
+		return
+	}
+	data.ID = types.StringValue(getResp.ID)
+	data.Name = types.StringValue(getResp.Name)
+	data.OrgID = types.StringValue(getResp.OrgID)
+	data.Type = types.StringValue(getResp.Type)
+	data.ClientID = types.StringValue(getResp.ClientID)
+	data.SecureConfigID = types.StringValue(getResp.SecureConfigID)
+	data.Status = types.StringValue(getResp.Status)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+func (r *AppOAuthResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan AppOAuthResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var state AppOAuthResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if plan.ID.IsNull() {
+		resp.Diagnostics.AddError(
+			"Invalid Resource ID",
+			"Cannot update OAuth provider without a valid ID.",
+		)
+		return
+	}
+
```

Comment:
We do not allow to change client ID. Should we do any validation it's still the same here?